sys/kern/tty.c

```diff
@@ -910,3 +911,31 @@ vnodeops_t tty_vnodeops = {
   .v_close = tty_vn_close,
   .v_getattr = tty_vn_getattr,
 };
+
+/* Controlling terminal pseudo-device (/dev/tty) */
+
+static int dev_tty_open(vnode_t *v, int mode, file_t *fp) {
+  proc_t *p = proc_self();
+  int error;
+
+  SCOPED_MTX_LOCK(all_proc_mtx);
+  tty_t *tty = p->p_pgrp->pg_session->s_tty;
+  if (!tty)
```

Comment:
Could you shortly explain this check?

Reply:
`/dev/tty` refers to the controlling terminal of the current process, if it has one. See https://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap10.html. I return `ENXIO` here because that's what NetBSD does: http://bxr.su/NetBSD/sys/kern/tty_tty.c#62

Reply:
I got it. Thanks